### PR TITLE
Makes flamethrowers easily refillable

### PR DESCRIPTION
**Pull request by TTerc**
[feature][qol]

## About the PR 

This PR lets you refill the flamethrower fuel tanks by simply clicking the large fuel tanks, water tanks, and other reagent dispensers.

## Why's this needed? 

Seeing people ditch their flamethrowers after they run out of fuel on blob rounds makes me sad. Hopefully it'll make flamethrowers a more viable weapon against blob and in general, as it makes refilling them way less tedious.

## Changelog

```changelog
(u)TTerc
(+)You can now refill the flamethrowers by simply clicking the large tanks with them.
```

### CODE REVIEW

File: code/obj/item/flamethrower.dm [hunk 1]
```diff
@@ -79,6 +79,23 @@ A Flamethrower in various states of assembly
 		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
 			return TRUE
 
+
+	/// allow refilling the fuel tank by simply clicking the reagent dispensers
+	afterattack(atom/target, mob/user, flag)
+		if(istype(target, /obj/reagent_dispensers) && get_dist(src,target) <= 1)
```
Comment (ZeWaka) [inline code/obj/item/flamethrower.dm:85]: `in_interact_range` check instead please

File: code/obj/item/flamethrower.dm [hunk 2]
```diff
@@ -79,6 +79,23 @@ A Flamethrower in various states of assembly
...
+			if(istype(src.fueltank))
+				var/obj/o = target
```
Comment (ZeWaka) [inline code/obj/item/flamethrower.dm:87]: either `O` or a different name.

File: code/obj/item/flamethrower.dm [hunk 3]
```diff
@@ -79,6 +79,23 @@ A Flamethrower in various states of assembly
...
+				o.reagents.trans_to(src.fueltank, (src.fueltank?.reagents?.maximum_volume - (src.fueltank?.reagents?.total_volume)))
+				inventory_counter?.update_percent(src.fueltank?.reagents?.total_volume, src.fueltank?.reagents?.maximum_volume)
```
Comment (ZeWaka) [inline code/obj/item/flamethrower.dm:89]: no need to nullcheck fueltank a million times, nor reagents - you already checked fueltank above and just wrap .reagents in an if/nif

File: code/obj/item/flamethrower.dm [hunk 4]
```diff
@@ -79,6 +79,23 @@ A Flamethrower in various states of assembly
...
+				boutput(user, "<span class='notice'>You refill the flamethrower's fuel tank.</span>")
+				playsound(src.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+				user.lastattacked = target
+				return
+			else
+				boutput(user, "<span class='notice'>Load the fuel tank first!</span>")
+				return
```
Comment (ZeWaka) [inline code/obj/item/flamethrower.dm:96]: useless returns

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by Azrun** [comment]
I think Action bar with a duration based on amount of fluid to transfer would be a great QoL with less of a balance change. Juggling disconnecting and reconnecting is not very fun.  Overall I like where this is going.

File: code/obj/item/flamethrower.dm (hunk 2 above)
Comment (TTerc) [reply code/obj/item/flamethrower.dm]: I think that works?

**Comment by TTerc** [comment]
> I think Action bar with a duration based on amount of fluid to transfer would be a great QoL with less of a balance change.

I tried messing with the actionbars but I feel it's out of the scope for me, at the moment. Actionbars could be added in a future PR if this gets merged and instant refill would become too much of an issue. But I'm kinda positive it shouldn't

**Comment by virvatuli-gh** [comment]
What's to stop someone from dragging around a welding fuel tank and essentially having a massive amount flamethrower juice? A single click instant-refill is pretty powerful and propels the flamethrower into being a better weapon than most other ranged options, you could just spam the flamethrower like this.

**Comment by TTerc** [comment]
> What's to stop someone from dragging around a welding fuel tank and essentially having a massive amount flamethrower juice? 

You can already do that if you have two small fuel tanks. Since flamer tanks can be easily swapped by clicking the tank. So if you wanted to run around with a welding fuel tank and spamming the flamer, nothing's stopping you.

Still, it takes one person dragging the tank away to stop the rampager.

**Comment by virvatuli-gh** [comment]
> You can already do that if you have two small fuel tanks. (which I don't think many people know about) Since flamer tanks can be easily swapped by clicking the flamer

You're right! I had no clue about that quickswap tank function. Consider my complaint addressed, there's no real difference between taking multiple small tanks or lugging a big one around then

File: code/obj/item/flamethrower.dm [hunk 5]
```diff
@@ -79,6 +79,20 @@ A Flamethrower in various states of assembly
 		if(istype(src.gastank) && src.gastank.air_contents && istype(src.fueltank) && src.fueltank.reagents)
 			return TRUE
 
+
+	/// allow refilling the fuel tank by simply clicking the reagent dispensers
+	afterattack(atom/target, mob/user, flag)
+		if(istype(target, /obj/reagent_dispensers) && in_interact_range(src,target))
```
Comment (Tarmunora) [inline code/obj/item/flamethrower.dm:85]: lol you could refill from ants on the floor

Comment (ZeWaka) [reply code/obj/item/flamethrower.dm:85]: lol

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**